store/store.go

```diff
@@ -22,13 +63,13 @@ type Storage interface {
 	Get(c cid.Cid) ([]IndexEntry, bool, error)
 	// Put stores a provider-piece entry for a CID if the entry is not already
 	// stored.  New entries are added to the entries that are already there.
-	Put(c cid.Cid, providerID peer.ID, pieceID cid.Cid) error
+	Put(c cid.Cid, entry IndexEntry) error
 	// PutMany stores the provider-piece entry for multiple CIDs
-	PutMany(cs []cid.Cid, providerID peer.ID, pieceID cid.Cid) error
+	PutMany(cs []cid.Cid, entry IndexEntry) error
 	// Remove removes a provider-piece entry for a CID
-	Remove(c cid.Cid, providerID peer.ID, pieceID cid.Cid) error
+	Remove(c cid.Cid, entry IndexEntry) error
 	// RemoveMany removes a provider-piece entry from multiple CIDs
-	RemoveMany(cids []cid.Cid, providerID peer.ID, pieceID cid.Cid) error
+	RemoveMany(cids []cid.Cid, entry IndexEntry) error
```

Comment:
*Not a blocker by any means, I can address it myself in my next PR, but wanted to open the discussion*

While working in the pass-through for the cache+persistence integration I realized that it may be convenient to add a `PutEntries/Move` (choose your preferred name) function for the cache so we can bulk put/move an `[]IndexEntry` from persistent to cache without having to loop through the entries putting them individually in the cache (which can be quite an overhead).

My suggestion would be to add:
```
type CacheStorage interface{
    Storage
     PutEntries(c cid.Cid, entries []IndexEntry) error
}
```
That way we have `StorageFlusher` (we maybe should consider changing the name) for the persistence layer, and `CacheStorage` for the cache.

(Although, I am happy to add `PutEntries` directly in `Storage` and implement it in all storages as it may also come handy for the persistence layer).

Reply:
Since you are already benchmarking this, it probably makes more sense for you to implement this so that you can tune it for maximum awesomeness.